ping/main.go

```diff
@@ -214,14 +216,15 @@ func runPing(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	// ☎️  Connect to all other peers.
 	//
 	// Note: we sidestep simultaneous connect issues by ONLY connecting to peers
-	// who published their addresses before us (this is enough to dedup and avoid
+	// that have a peerID smaller to our (this is enough to dedup and avoid
```

Comment:
As far as I understand this case can not happen because we never append ourselves (see line ~173)

Reply:
We never connect to our self. In addition, by `break`ing early we also prevent two nodes to both connect to each other, thus resulting in two connections between each pair of peers. Does that make sense @laurentsenta?

Reply:
That's the intent of the code,
but as far as I understand we never add ourselves to the `peers` array:

Note the `continue` when we find ourselves, at construction time:
https://github.com/libp2p/test-plans/blob/8497c05980adb9651aad78b15d515bf97f947434/ping/main.go#L167-L178

So later on, there is no item that verifies `ai.ID == id` => We don't dedup the connections.